src/training/finetune.py

```diff
@@ -0,0 +1,234 @@
+        "--precision",
+        choices=["amp", "amp_bf16", "amp_bfloat16", "bf16", "fp16", "fp32"],
+        default="amp",
+        help="Floating point precision."
+    )
+    parser.add_argument(
+        "--model",
+        type=str,
+        default="RN50",
+        help="Name of the vision backbone to use.",
+    )
+    parser.add_argument(
+        "--pretrained",
+        default='',
+        type=str,
+        help="Use a pretrained CLIP model weights with the specified tag or file path.",
+    )
+    parser.add_argument(
+        "--seed", type=int, default=0, help="Default random seed."
+    )
+
+    args = parser.parse_args(args)
+    return args
+
+
+class GLUEDataset(Dataset):
+    """GLUE dataset."""
+
+    def __init__(self, task, split, text_key, label_key, tokenizer=None):
+        super().__init__()
+
+        try:
+            from datasets import load_dataset
+        except ImportError:
+            raise ImportError("Please install HF datasets: pip install datasets")
+
+        self.dataset = load_dataset("glue", task, split=split)
+        self.label_key = label_key
+        self.text_key = text_key
+        self.length = len(self.dataset)
+        self.tokenize = tokenizer
+
+    def __len__(self):
+        return self.length
+
+    def __getitem__(self, idx):
+        item = self.dataset[idx]
+        text = item[self.text_key]
+        label = item[self.label_key]
+        return {
+            'text': self.tokenize([text])[0],
+            'label': label
+        }
+
+
+class SequenceClassifier(nn.Module):
+    def __init__(self, encoder, embed_dim, num_labels):
+        super().__init__()
+
+        self.encoder = encoder
+        self.logits_proj = nn.Linear(embed_dim, num_labels)
+
+    def forward(self, text):
+        text_features = self.encoder.encode_text(text)
+        logits = self.logits_proj(text_features)
+        return logits
+
+
+def get_task_metric(task_name):
+    if task_name == "cola":
+        metric = evaluate.load("glue", "stsb")
+    else:
+        metric = evaluate.load("glue", task_name)
+    return metric
+
+
+def get_task_dataloaders(args):
+    tokenizer = get_tokenizer(args.model)
+    task_name = args.task_name
+
+    dataloaders = {}
+    for split_name in ["train", "validation", "test"]:
+        dataset = GLUEDataset(
+            task_name,
+            split_name,
+            text_key="sentence",
+            label_key="label",
+            tokenizer=tokenizer,
+        )
+        dataloader = DataLoader(
+            dataset,
+            batch_size=args.batch_size,
+            shuffle=True,
+            num_workers=args.workers,
+            pin_memory=True,
+            drop_last=True,
+        )
+        dataloaders[split_name] = dataloader
+
+    return dataloaders
+
+
+def compute_metrics(model, dataloader, device, args):
+    model.eval()
+    metric = get_task_metric(args.task_name)
+    val_loss = 0
+    samples_seen = 0
+    import pdb; pdb.set_trace()
```

Comment:
Leftover? Do you want to in general check for these and make sure you don't land rouge code here?